src/openrct2/world/Map.cpp

```diff
@@ -941,27 +945,8 @@ bool map_is_location_at_edge(const CoordsXY& loc)
  *
  *  rct2: 0x0068B280
  */
-void tile_element_remove(TileElement* tileElement)
-{
-    // Replace Nth element by (N+1)th element.
-    // This loop will make tileElement point to the old last element position,
-    // after copy it to it's new position
-    if (!tileElement->IsLastForTile())
-    {
-        do
-        {
-            *tileElement = *(tileElement + 1);
-        } while (!(++tileElement)->IsLastForTile());
-    }
-
-    // Mark the latest element with the last element flag.
-    (tileElement - 1)->SetLastForTile(true);
-    tileElement->base_height = MAX_ELEMENT_HEIGHT;
-
-    if ((tileElement + 1) == gNextFreeTileElement)
-    {
-        gNextFreeTileElement--;
-    }
+void tile_element_remove(TileElement* tileElement) {
+    tileElement->Remove();
```

Comment:
I moved this logic into `TileElement::Remove()` as it seemed to make more sense to live in there rather than this decompiled function. Does this make sense?